tests/components/influxdb/test_init.py

```diff
@@ -131,6 +131,126 @@ async def test_setup_config_full(hass, mock_client, config_ext, get_write_api):
     assert get_write_api(mock_client).call_count == 1
 
 
+@pytest.mark.parametrize(
+    "mock_client, config_ext",
+    [
+        (
+            influxdb.DEFAULT_API_VERSION,
+            {
+                "ssl": True,
+                "verify_ssl": False,
+            },
+        ),
+        (
+            influxdb.DEFAULT_API_VERSION,
+            {
+                "ssl": True,
+                "verify_ssl": True,
+            },
+        ),
+        (
+            influxdb.DEFAULT_API_VERSION,
+            {
+                "ssl": True,
+                "verify_ssl": True,
+                "ssl_ca_cert": "fake/path/ca.pem",
+            },
+        ),
+        (
+            influxdb.DEFAULT_API_VERSION,
+            {
+                "ssl": True,
+                "ssl_ca_cert": "fake/path/ca.pem",
+            },
+        ),
+        (
+            influxdb.DEFAULT_API_VERSION,
+            {
+                "ssl": True,
+                "verify_ssl": False,
+                "ssl_ca_cert": "fake/path/ca.pem",
+            },
+        ),
+        (
+            influxdb.API_VERSION_2,
+            {
+                "api_version": influxdb.API_VERSION_2,
+                "verify_ssl": False,
+            },
+        ),
+        (
+            influxdb.API_VERSION_2,
+            {
+                "api_version": influxdb.API_VERSION_2,
+                "verify_ssl": True,
+            },
+        ),
+        (
+            influxdb.API_VERSION_2,
+            {
+                "api_version": influxdb.API_VERSION_2,
+                "verify_ssl": True,
+                "ssl_ca_cert": "fake/path/ca.pem",
+            },
+        ),
+        (
+            influxdb.API_VERSION_2,
+            {
+                "api_version": influxdb.API_VERSION_2,
+                "verify_ssl": False,
+                "ssl_ca_cert": "fake/path/ca.pem",
+            },
+        ),
+    ],
+    indirect=["mock_client"],
+)
+async def test_setup_config_ssl(hass, mock_client, config_ext):
+    """Test the setup with various verify_ssl values."""
+    config = {
+        "influxdb": BASE_V2_CONFIG
+        if config_ext.get("api_version") == influxdb.API_VERSION_2
+        else BASE_V1_CONFIG
+    }
+    config["influxdb"].update(config_ext)
+
+    with patch("os.access", return_value=True):
+        with patch("os.path.isfile", return_value=True) as mock_isfile:
+            assert await async_setup_component(hass, influxdb.DOMAIN, config)
+            await hass.async_block_till_done()
+
+            assert hass.bus.listen.called
+            assert EVENT_STATE_CHANGED == hass.bus.listen.call_args_list[0][0][0]
+
+            if config_ext.get("api_version") == influxdb.API_VERSION_2:
+                for arg in ("ssl_ca_cert", "verify_ssl"):
+                    if arg in config_ext:
```

Comment:
Can we include the expected call arguments in the parameters so we can avoid the logic in the test?